src/Resources/app/administration/src/module/sw-extension-store/service/extension-store.service.ts

```diff
@@ -0,0 +1,51 @@
+import type { ShopwareDiscountCampaignService } from 'src/app/service/discount-campaign.service';
+import type { ExtensionVariant } from 'src/module/sw-extension/service/extension-store-action.service';
+import type { ShopwareExtensionService } from 'src/module/sw-extension/service/shopware-extension.service';
+
+export default class ExtensionStoreService {
+    constructor(
+        private readonly discountCampaignService: ShopwareDiscountCampaignService,
+        private readonly extensionService: ShopwareExtensionService,
+    ) {
+    }
+
+    public orderVariantsByPricePerMonth(variants: ExtensionVariant[]): ExtensionVariant[] {
+        return variants.sort((first, second) => {
+            const firstPrice = this.getPriceFromVariant(first, true);
+            const secondPrice = this.getPriceFromVariant(second, true);
+
+            return firstPrice - secondPrice;
+        });
+    }
+
+    public getPriceFromVariant(variant: ExtensionVariant, perMonth = false): number {
+        if (this.isVariantDiscounted(variant)) {
+            // null assertion is fine here because we do all checks in isVariantDiscounted
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            return perMonth ? variant.discountCampaign!.discountedPricePerMonth! : variant.discountCampaign!.discountedPrice!;
+        }
+
+        return perMonth ? variant.netPricePerMonth : variant.netPrice;
+    }
+
+    public isExtensionDiscounted(variants: ExtensionVariant[]): boolean {
+        return variants.some((variant) => this.isVariantDiscounted(variant));
+    }
+
+    public isVariantDiscounted(variant: ExtensionVariant): boolean {
```

Comment:
I think you could even do something like this to get rid of the `eslint-disable-next-line` in the other method. Correct me if I'm wrong, though.

```suggestion
    public isVariantDiscounted(variant: ExtensionVariant): variant is ExtensionVariant & {
        discountCampaign: {
            discountedPrice: number;
            discountedPricePerMonth: number;
        };
    } {
```

Reply:
Will evalute this and report back here...

Reply:
Looks good, thanks for the tip!